thunder/torch/__init__.py

```diff
@@ -146,6 +146,7 @@ def __init__(
         is_prim: bool = False,
         tags: None | list[Any] = None,
         out_of_place: Symbol | None = None,
+        allow_only_tensorproxy: bool = True,
```

Comment:
I personally prefer something like `allow_tensor_subclass_proxy: bool = False` to `allow_only_tensorproxy: bool = True`.
This is because it'd be a bit clearer in that there exists some sort of support of tensor wrapper subclasses.